### ISSUE TITLE
Keepalive thread's  interrupt signal may be catched when it is sleeping

### ISSUE THREAD

**Issue by gatesking** [hierynomus/sshj#283]
In Keepalive.java:
```
try {
            while (!isInterrupted()) {
                final int hi = getPositiveInterval();
                if (conn.getTransport().isRunning()) {
                    log.debug("Sending keep-alive since {} seconds elapsed", hi);
                    doKeepAlive();
                }
                **Thread.sleep(hi * 1000);**
            }
        } catch (Exception e) {
            // If we weren't interrupted, kill the transport, then this exception was unexpected.
            // Else we're in shutdown-mode already, so don't forcibly kill the transport.
            if (!isInterrupted()) {
                conn.getTransport().die(e);
            }
        }
```
when keepalive interrupt is called in ConnectionImpl.notifyError, the interrupt signal may be catched by Thread.sleep, as a result, it will still go into conn.getTransport().die(e).
I suggest catch interrupt signal will be better, am I right?
```
try {
            while (!isInterrupted()) {
                final int hi = getPositiveInterval();
                if (conn.getTransport().isRunning()) {
                    log.debug("Sending keep-alive since {} seconds elapsed", hi);
                    doKeepAlive();
                }
                Thread.sleep(hi * 1000);
            }
        } **catch (InterruptedException e) {
                // do nothing
        }** catch (Exception e) {
            // If we weren't interrupted, kill the transport, then this exception was unexpected.
            // Else we're in shutdown-mode already, so don't forcibly kill the transport.
            if (!isInterrupted()) {
                conn.getTransport().die(e);
            }
        }
```
Thanks!

**Comment by hierynomus** [comment]
Yes, you're right. Please send a PR

2016-11-29 15:09 GMT+01:00 gatesking <notifications@github.com>:

**Comment by hierynomus** [comment]
Merged #284, thanks!
